runtime/compiler/runtime/JITServerIProfiler.cpp

```diff
@@ -803,8 +799,11 @@ JITClientIProfiler::serializeIProfilerMethodEntry(TR_OpaqueMethodBlock *omb)
    auto entry = findOrCreateMethodEntry(NULL, (J9Method *) omb, false);
    if (entry)
       {
-      auto serialEntry = TR_ContiguousIPMethodHashTableEntry::serialize(entry);
-      std::string entryStr((char *) &serialEntry, sizeof(TR_ContiguousIPMethodHashTableEntry));
+      std::string entryStr;
+      entryStr.reserve(sizeof(TR_ContiguousIPMethodHashTableEntry));
```

Comment:
`reserve` will not work here because the size of the string is not adjusted. We need to do `std::string entryStr(sizeof(TR_ContiguousIPMethodHashTableEntry), 0)`
An alternative is to resize the string before calling `serialize`: 
entryStr.resize(sizeof(TR_ContiguousIPMethodHashTableEntry))` which fills it with 0 as well

We could also another `resize` after the string is filled because we know how many entries are stored. This will limit the amount of empty data transferred to and from network.

Reply:
Ok, used a different constructor